frontend/lib/resolution/scope-queries.cpp

```diff
@@ -845,19 +861,40 @@ findUseImportTarget(Context* context,
                     UniqueString& foundName) {
   if (auto ident = expr->toIdentifier()) {
     foundName = ident->name();
+
     if (ident->name() == USTR("super")) {
-      return scope->parentScope()->moduleScope();
+      auto ret = handleSuperMaybeError(context,
+                                       scope, // scopeToAdvanceFrom
+                                       scope, // scopeForOrigin
+                                       expr,
+                                       useOrImport);
+      return ret;
     } else if (ident->name() == USTR("this")) {
       return scope->moduleScope();
     } else {
       return findScopeViz(context, scope, ident->name(), resolving,
                           expr->id(), useOrImport, /* isFirstPart */ true);
     }
   } else if (auto dot = expr->toDot()) {
-    UniqueString ignoredFoundName;
+    UniqueString ignoreFoundName;
     const Scope* innerScope = findUseImportTarget(context, scope, resolving,
                                                   dot->receiver(), useOrImport,
-                                                  ignoredFoundName);
+                                                  ignoreFoundName);
+    // TODO: 'this.this'?
+    if (dot->field() == USTR("super")) {
+
+      // TODO: In production this case is the same as the above, but maybe
+      // they ought to be different. Technically, this isn't trying to
+      // use 'super' from a top-level module - it's actually just adding
+      // one too many 'super' to the chain.
+      auto ret = handleSuperMaybeError(context,
+                                       innerScope, // scopeToAdvanceFrom
+                                       scope,      // scopeForOrigin
```

Comment:
I don't see why we wouldn't always use `innerScope` for reporting the 'super' error. As things stand, if you do:
```Chapel
module M1 {
  module M2 {
    import super.super;
  }
}
```
Won't it use `M2` as scope, and print "`M2` is declared at the top level here"?

Reply:
I think you're right, so for now I'll leave that argument ignored. I think eventually we'll need the two different scopes for the TODO, though.

Reply:
Nope `{}` will work, just missed a spot 😄.